File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
       int count = 0, best = Integer.MIN_VALUE;
       for (int num : nums) {
         count += num;
         best = Math.max(best, count);
         if (count < 0) {
            count = 0;
         }
       }
       return best;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_val, curr_max = nums[0], 0
        for num in nums:
            curr_max += num
            max_val = max(curr_max, max_val)
            if curr_max < 0: curr_max = 0
        return max_val